src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -94,7 +86,13 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             final AdaptedAddressBook toSave = new AdaptedAddressBook(addressBook);
             final Marshaller marshaller = jaxbContext.createMarshaller();
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
-            marshaller.marshal(toSave, fileWriter);
+            if (isEncrypted) {
```

Comment:
I feel that the contents in this `if` block should be swapped with the contents in `else` block. Then when the program checks `if (isEncrypted)` and it returns true, the file would be saved. `else` it would perform the encryption before saving.

You could write a separate function to check if the file is encrypted and call it in `load()` and that function will set `isEncrypted` to true if the file is encrypted.